### PR TITLE
Tribes Townsquare

### PR DESCRIPTION
**Pull request by androolloyd**
##### Description

Tribes Townsquare Integrated View

##### Refers/Fixes
#6495

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@androolloyd yo we got conflict

File: docker-compose.yml [hunk 1]
```diff
@@ -81,19 +81,6 @@ services:
         reservations:
           memory: 512M
 
-  testrpc:
```
Comment (thelostone-mc) [inline docker-compose.yml:84]: oh niceee

Comment (androolloyd) [reply docker-compose.yml:84]: I can undo this, do we even use testrpc for anything? I do all my testing to rinkeby, just eats up ram if we're not using it actively.

File: app/dashboard/views.py [hunk 2]
```diff
@@ -915,25 +915,33 @@ def projects_fetch(request):
...
 
-    try:
-        hackathon_event = HackathonEvent.objects.get(id=hackathon_id)
-    except HackathonEvent.DoesNotExist:
-        hackathon_event = HackathonEvent.objects.last()
+    if hackathon_id:
+        try:
+            hackathon_event = HackathonEvent.objects.get(id=hackathon_id)
+        except HackathonEvent.DoesNotExist:
+            hackathon_event = HackathonEvent.objects.last()
```
Comment (danlipert) [inline app/dashboard/views.py:922]: Is this a fallback for some other page that users this endpoint?

File: app/dashboard/views.py [hunk 3]
```diff
@@ -1002,8 +1012,22 @@ def users_fetch(request):
...
-    show_banner = None
+    if tribe:
+
+        profile_list = profile_list.filter(Q(follower__org__handle__in=[tribe]) | Q(organizations_fk__handle__in=[tribe]))
+
+        if user_filter and user_filter != 'all':
+            if user_filter == 'owners':
+                profile_list = profile_list.filter(Q(organizations_fk__handle__in=[tribe]))
+
+            if user_filter == 'members':
```
Comment (danlipert) [inline app/dashboard/views.py:1023]: should be elif here

File: app/dashboard/views.py [hunk 4]
```diff
@@ -2810,18 +2838,29 @@ def profile(request, handle, tab=None):
     context['tab'] = tab
     context['show_activity'] = request.GET.get('p', False) != False
     context['is_my_org'] = request.user.is_authenticated and any([handle.lower() == org.lower() for org in request.user.profile.organizations ])
-    context['is_on_tribe'] = False
-    if request.user.is_authenticated:
-        context['is_on_tribe'] = request.user.profile.tribe_members.filter(org__handle=handle.lower())
+    if request.user.is_authenticated and hasattr(request.user, 'profile'):
+        context['is_on_tribe'] = len(request.user.profile.tribe_members.filter(org__handle=handle.lower())) > 0
```
Comment (danlipert) [inline app/dashboard/views.py:2842]: can use `exists()` here rather than checking the length I think

File: docker-compose.yml (hunk 1 above)
Comment (danlipert) [reply docker-compose.yml:84]: yeah lets ditch it if possible - but I want to make sure @octavioamu and @owocki don't use it either - can you put this in a separate PR?

File: app/dashboard/views.py (hunk 2 above)
Comment (androolloyd) [reply app/dashboard/views.py:922]: I believe it’s a fallback for other requests

File: app/assets/v2/js/pages/profile-tribes.js [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+  }
+
+  if (existingScript && callback)
+    callback();
+};
+
+(function($) {
+  // doc ready
+  $(() => {
+    Vue.use(VueQuillEditor);
```
Comment (octavioamu) [inline app/assets/v2/js/pages/profile-tribes.js:27]: Is this loaded?

File: app/assets/v2/js/pages/profile-tribes.js [hunk 6]
```diff
@@ -0,0 +1,272 @@
+/* eslint-disable no-loop-func */
+
+
+const loadDynamicScript = (callback, url, id) => {
```
Comment (octavioamu) [inline app/assets/v2/js/pages/profile-tribes.js:4]: this is already on tribe-edit.js also loaded on profile so you probably have it already on that page

File: app/dashboard/views.py [hunk 7]
```diff
@@ -4499,13 +4546,18 @@ def join_tribe(request, handle):
     if request.user.is_authenticated:
         profile = request.user.profile if hasattr(request.user, 'profile') else None
         try:
-            TribeMember.objects.get(profile=profile, org__handle=handle.lower()).delete()
+            try:
+                TribeMember.objects.get(profile=profile, org__handle=handle.lower()).delete()
+            except TribeMember.MultipleObjectsReturned:
+                TribeMember.objects.filter(profile=profile, org__handle=handle.lower()).delete()
+
```
Comment (octavioamu) [inline app/dashboard/views.py:4553]: should be situations where profile return multiple? and in that case do we want to delete all of them?

File: package-lock.json [hunk 8]
```diff
@@ -3670,6 +3670,11 @@
       "integrity": "sha1-IZMqVJ9eUv/ZqCf1cOBL5iqX2lQ=",
       "dev": true
     },
+    "prettier": {
```
Comment (octavioamu) [inline package-lock.json:3673]: is this change package related?

File: package.json [hunk 9]
```diff
@@ -25,5 +25,8 @@
     "eslint": "^5.16.0",
     "stylelint": "^9.10.1",
     "stylelint-config-recommended": "^2.2.0"
+  },
+  "dependencies": {
+    "prettier": "^2.0.5"
```
Comment (octavioamu) [inline package.json:30]: is this being used do we need to add a new dependency as we don't really run node only manually when checking lint.?

File: app/assets/v2/js/pages/profile-tribes.js (hunk 5 above)
Comment (androolloyd) [reply app/assets/v2/js/pages/profile-tribes.js:27]: i believe is loaded by the profile-vue.html

File: package.json (hunk 9 above)
Comment (androolloyd) [reply package.json:30]: ah nah that can be removed.

File: app/dashboard/views.py (hunk 7 above)
Comment (androolloyd) [reply app/dashboard/views.py:4553]: shouldn't ever be multiple members for a single profile and an org.

Comment (androolloyd) [reply app/dashboard/views.py:4553]: the issue is that if a request fails, the user could get multiple entries in the db and then can't follow/unfollow properly, this ensures that we delete any multiple entries and keep the user's ability to follow/unfollow in tact.

File: app/assets/v2/js/pages/profile-tribes.js (hunk 6 above)
Comment (androolloyd) [reply app/assets/v2/js/pages/profile-tribes.js:4]: i don't use tribe-edit and am not actually using the load dynamic script yet, i just hard coded on to the tribes-vue, lets leave as I intend to refactor to use this.
